core/src/main/java/org/fao/geonet/kernel/ThesaurusManager.java

```diff
@@ -397,6 +394,20 @@ public Thesaurus getThesaurusByConceptScheme(String uri) {
         return null;
     }
 
+    public Thesaurus getThesaurusByTitle(String title) {
+        for (Map.Entry<String, Thesaurus> entry : getThesauriMap().entrySet()) {
+            try {
+                Thesaurus thesaurus = entry.getValue();
+                if (thesaurus.getTitle().equals(title)) {
```

Comment:
Can be relevant to check in `Thesaurus.getMultilingualTitles` also if contains values?

Reply:
I've some improvements related to multilingual thesaurus title support planned. We could improve that at that time I think.

Reply:
Sounds good